doc/testing.md

```diff
@@ -0,0 +1,158 @@
+# testing
+
+Automated testing is an essential tool to assure correctness.
+
+## Scopes
+
+The testing strategy for polkadot is 4-fold:
+
+### Unit testing
+
+Boring, small scale correctness tests of individual functions.
+
+### Integration tests
+
+One particular subsystem (subsystem under test) interacts with a
+mocked overseer that is made to assert incoming and outgoing messages
+of the subsystem under test.
```

Comment:
This sounds kind of like what we do with our unit tests of subsystems. Could you clarify the difference?

Reply:
It pretty much is. There is little consistency between the subsystems and some generalization with addition helper macros might reduce the boiler plate a bit. That's the only thing left for this point and is certainly of low priority.

Reply:
So you make a distinction between integration test and system tests. It seems like this section is redundant

Reply:
They are two distinct things:

a `integration test` is (in our case) the test of exactly one individual subsystem w/ mocked surroundings (overseer, other subsystems) where as a `system test` is the test of a node w/ partially real, partially mocked subsystems in conjunction w/ other subsystems. I guess some additional bike-shedding is needed to make the distinction cleaner.